### DISCUSSION TITLE
[Q] Add validation to all cells of a column

### DISCUSSION THREAD

**Discussion by ghost** [exceljs/exceljs#2406] [Q&A]
## 💬 Questions and Help

Hi,
I want to know if there is a way to put validation to all cells of given column instead of just one cell? Pardon me if I am missing something but I went through the documentation but couldn't find it.

I tried following way to do that but it is not working for me when I am opening in LibreOffice Calc.

```
let index = 3;

const dobCol = worksheet.getColumn(index);
dobCol.header = 'Date of Birth';
dobCol.eachCell({ includeEmpty: true }, function (cell, rowNumber) {
    cell.dataValidation = {
        type: 'list',
        allowBlank: false,
        formulae: ['"a,b,c"']
    };
});
```
The above method is only putting validation to the first cell of that column.

Thanks a lot for this library.

**Comment by seompark** [comment]
Any update?

**Comment by ankuratudemy** [comment]
Did you find solution to this ?
Seems like we need to make dbCol.eachCell  loop to be synchronous.  excel gets created before the loop ends.

**Comment by wangchao18654941093** [comment]
You can try to solve the problem in the following ways:
ws.dataValidations.add("X2:X9999",{
    type: 'list',
    allowBlank: true,
    formulae: ['"male,female,other"']
  });

**Comment by wangchao18654941093** [comment]
#1109
